cpp/src/arrow/ipc/ipc-adapter-test.cc

```diff
@@ -91,13 +91,21 @@ TEST_F(TestWriteRowBatch, IntegerRoundTrip) {
   int64_t header_location;
   ASSERT_OK(WriteRowBatch(mmap_.get(), &batch, 0, &header_location));
 
+  MockMemorySource mock_source(1 << 16);
```

Comment:
It probably pays to move the verification of GetRowBatchSize to a separate unit test.